tss-esapi/src/context/tpm_commands/attestation_commands.rs

```diff
@@ -1,18 +1,142 @@
 // Copyright 2021 Contributors to the Parsec project.
 // SPDX-License-Identifier: Apache-2.0
 use crate::{
-    handles::KeyHandle,
-    structures::{Data, PcrSelectionList},
-    tss2_esys::*,
+    handles::{KeyHandle, ObjectHandle},
+    structures::{AttestBuffer, Data, PcrSelectionList, SignatureScheme},
+    tss2_esys::{Esys_Certify, Esys_Quote, TPM2B_ATTEST, TPMT_SIG_SCHEME},
     utils::Signature,
     Context, Error, Result,
 };
 use log::error;
 use mbox::MBox;
+use std::convert::TryFrom;
 use std::ptr::null_mut;
 
 impl Context {
-    // Missing function: Certify
+    /// Certify that an object is resident in the TPM.
+    ///
+    /// # Arguments
+    /// * `object_handle` - The object to be certified
+    /// * `signing_key_handle` - The key to be used for signing the
+    /// attestation structure.
+    /// * `qualifying_data` - Data provided by the caller, generally
+    /// to ensure freshness of the attestation.
+    /// * `scheme` - Signature scheme to be used with the signing key.
+    ///
+    /// # Example
+    /// ```rust,no_run
```

Comment:
Yeah, I wanted it to be compiled only - otherwise I'd have to actually generate all the keys which is a massive pain. The compilation should at least make sure that the example has valid syntax